api/types/types.proto

```diff
@@ -366,6 +368,27 @@ message GCPCloudSQL {
 message Azure {
     // Name is the Azure database server name.
     string Name = 1 [ (gogoproto.jsontag) = "name,omitempty" ];
+    // Region is an Azure cloud region.
+    string Region = 2 [ (gogoproto.jsontag) = "region,omitempty" ];
+    // ResourceID is the Azure fully qualified ID for the resource.
+    AzureResourceID ResourceID = 3
```

Comment:
nit: ResourceID name indicate that this will be some kind of uniq identifier but instead this filed contains struct with other field that makes a resourceID so would would you say about renaming this field to `AzureResource`

Reply:
Actually, I think for future-proof we should just make this ResourceID a string. I like keeping the fields separate for ease of use and readability, but as I looked at the `arm.ResourceID` definition I realized that the ID can be nested.

Reply:
The `armmysql.Server` and `armpostgresql.Server` have comments that imply the ID can only be of the form `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}` though. I'm guessing that the nested form doesn't apply to databases then - what do you guys think?

Reply:
I think the current way is fine once the struct name is updated. If one day we run into the nested ID, it can be dealt with separately.

Though I think the real question is what we need these values for, after `types.Database` is created. I checked the big PR I don't see any use for it yet.

Reply:
I needed the resource group for making the metadata watcher, since `armmysql.ServersClient` and `armpostgresql.ServersClient` require the resource name and group to fetch a resource. The resource group is only specified in the resource ID string, which requires parsing. I added all the relevant fields from parsing the ID so I could just parse once, but the only one I need so far is that resource group. I also added the region, but this has no use right now either.

All that being said I'm not sure if we even have a use-case for metadata watcher service for Azure. We do need the server to have an Azure Active Directory admin set to enable AAD auth. If a user ever unsets the AD admin on the database server then AAD is effectively disabled. So this metadata watcher could act like it does for RDS and detect if AAD auth is disabled. I'm just not sure if that's useful since a user must still login manually and create the user/role in the database identified by the service principal ID - and if we "reenable" AAD auth by setting an AD admin, then we would have to set the admin to the service principal itself...

Reply:
The metadata service is not designed as a watcher currently. The metadata service is only run once per new database (with the exception of ElastiCache and MemoryDB but you can ignore). I believe the metadata service was mainly designed to fetch metadata for non-auto-discovered databases (e.g. from static files), since auto-discovered databases usually come with proper metadata. The auto-discovery watchers should detect any metadata change and the reconciler can push a new database. 

Do we have a plan on "auto" enabling AAD auth? Maybe we can revisit this once we work on that. 

Is anything else requiring the ResourceID apart from the metadata service?

Reply:
I don't think anything else needs the resource ID. I'll just hold off on changing the protobuf if we want to revisit this in the future. I think we should eventually automate the creation of a role/user in the database itself, but we don't do that for AWS right now either.